### PR TITLE
HttpServerRequest throws error when setExpectMultipart fails.

### PR DESCRIPTION
**Pull request by Warkst**
This commit implements the suggestion provided in issue #2960.
The functionality is implemented for both HttpServerRequestImpl and
Http2ServerRequestImpl. There is also a new set of tests included.

### CODE REVIEW

**Comment by vietj** [comment]
can you avoid reformating imports ?

**Comment by Warkst** [comment]
I turned off my IDE's auto optimise settings and updated the PR, is it not okay now?

**Comment by vietj** [comment]
now we need to review it :-)

File: src/main/java/io/vertx/core/http/impl/Http2ServerRequestImpl.java [hunk 1]
```diff
@@ -11,13 +11,7 @@
...
 package io.vertx.core.http.impl;
 
-import io.netty.handler.codec.http.DefaultHttpContent;
-import io.netty.handler.codec.http.DefaultHttpRequest;
-import io.netty.handler.codec.http.HttpHeaderNames;
-import io.netty.handler.codec.http.HttpHeaderValues;
-import io.netty.handler.codec.http.HttpRequest;
-import io.netty.handler.codec.http.HttpUtil;
-import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.handler.codec.http.*;
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/Http2ServerRequestImpl.java]: remove import change

File: src/main/java/io/vertx/core/http/impl/Http2ServerRequestImpl.java [hunk 2]
```diff
@@ -27,16 +21,9 @@
...
-import io.vertx.core.http.HttpConnection;
 import io.vertx.core.http.HttpMethod;
-import io.vertx.core.http.HttpServerFileUpload;
-import io.vertx.core.http.HttpServerRequest;
 import io.vertx.core.http.HttpVersion;
-import io.vertx.core.http.ServerWebSocket;
-import io.vertx.core.http.StreamPriority;
-import io.vertx.core.http.StreamResetException;
-import io.vertx.core.http.HttpFrame;
+import io.vertx.core.http.*;
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/Http2ServerRequestImpl.java]: remove import change

**Comment by vietj** [comment]
I still see lot of import changes

**Comment by Warkst** [comment]
Oh, it looks like I forgot to undo the changes in Http2ServerRequestImpl. I've added another commit to the PR that should correct this.

File: src/main/java/io/vertx/core/http/impl/Http2ServerRequestImpl.java [hunk 3]
```diff
@@ -479,7 +479,11 @@ public HttpServerRequest setExpectMultipart(boolean expect) {
                   uri);
               req.headers().add(HttpHeaderNames.CONTENT_TYPE, contentType);
               postRequestDecoder = new HttpPostRequestDecoder(new NettyFileUploadDataFactory(context, this, () -> uploadHandler), req);
+            } else {
+                throw new IllegalStateException("Invalid content type or http method. Content type should be "+HttpHeaderValues.MULTIPART_FORM_DATA.toString()+" or "+HttpHeaderValues.APPLICATION_X_WWW_FORM_URLENCODED+" and http method should be "+ io.netty.handler.codec.http.HttpMethod.POST.toString()+", " + io.netty.handler.codec.http.HttpMethod.PUT.toString() + ", "+ io.netty.handler.codec.http.HttpMethod.PATCH.toString()+" or "+io.netty.handler.codec.http.HttpMethod.DELETE.toString()+".");
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/Http2ServerRequestImpl.java]: can you just write POST, PUT, PATCH, DELETE ? that's easier to read and those are well known constants that don't need to use Java constants.

Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/Http2ServerRequestImpl.java]: same for content types

**Comment by vietj** [comment]
can you sign your commits ? they don't pass the eclipse validation

**Comment by Warkst** [comment]
I've changed the exception message as you requested and made sure I signed the commit. I didn't notice until after I pushed that my IDE removed some whitespace characters, I hope you don't mind...

**Comment by vietj** [comment]
if you don't sign your commits we cannot accept the PR @Warkst can you submit a new PR with commit signed, ECA signed ?

**Comment by vietj** [comment]
see https://github.com/eclipse-vertx/vert.x/commit/ed0b65bf7c5daed2d53336f0a01a454f7efc9884

**Pull request closed without merging**